### PR TITLE
[3.5] Add new lwm2mObjectModels to install

### PR DESCRIPTION
**Pull request by nickAS21**
## Pull Request description

add new lwm2mObjectModels to install
- источник: application/src/main/data/lwm2m-registry (название и файлы отсюда: https://github.com/OpenMobileAlliance/lwm2m-registry)
процедуру вызываем из: application/src/main/java/org/thingsboard/server/install/ThingsboardInstallService.java -> case "3.4.4":
- tenant -> TenantId.SYS_TENANT_ID

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] PR name contains fix version. For example, "[3.3.4] Hotfix of some UI component" or "[3.4] New Super Feature".
- [x] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Required for internal contributors only.

## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [x] If new dependency was added: the dependency tree is checked for conflicts.
- [x] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [x] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.

### CODE REVIEW

File: application/src/main/java/org/thingsboard/server/service/install/InstallScripts.java [hunk 1]
```diff
@@ -260,4 +266,53 @@ public void createOAuth2Templates() throws Exception {
             );
         }
     }
+
+    /**
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/service/install/InstallScripts.java:270]: Plz remove commented code.

File: application/src/main/java/org/thingsboard/server/service/install/InstallScripts.java [hunk 2]
```diff
@@ -260,4 +266,53 @@ public void createOAuth2Templates() throws Exception {
...
+     */
+    public void migrateLwm2mResources() {
+        Path resourceLwm2mPath = Paths.get(dataDir, MODELS_LWM2M_DIR);
+        try (DirectoryStream<Path> dirStream = Files.newDirectoryStream(resourceLwm2mPath, path -> path.toString().endsWith(InstallScripts.XML_EXT))) {
+            dirStream.forEach(
+                    path -> {
+                        try {
+                            String data = Base64.getEncoder().encodeToString(Files.readAllBytes(path));
+                            TbResource tbResource = new TbResource();
+                            tbResource.setTenantId(TenantId.SYS_TENANT_ID); // SisAdmin
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/service/install/InstallScripts.java:284]: remove `// SisAdmin`

File: application/src/main/java/org/thingsboard/server/install/ThingsboardInstallService.java [hunk 3]
```diff
@@ -88,6 +90,12 @@ public class ThingsboardInstallService {
     @Autowired(required = false)
     private TsLatestMigrateService latestMigrateService;
 
+    @Autowired
+    private ResourceService resourceService;
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/install/ThingsboardInstallService.java:94]: not used

File: application/src/main/java/org/thingsboard/server/install/ThingsboardInstallService.java [hunk 4]
```diff
@@ -284,6 +293,7 @@ public void performInstall() {
                 systemDataLoaderService.createQueues();
 //                systemDataLoaderService.loadSystemPlugins();
 //                systemDataLoaderService.loadSystemRules();
+                installScripts.migrateLwm2mResources();
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/install/ThingsboardInstallService.java:296]: better to use systemDataLoaderService for loading lwm2m resources
`systemDataLoaderService.loadSystemResources();`

File: application/src/main/java/org/thingsboard/server/service/install/InstallScripts.java [hunk 5]
```diff
@@ -260,4 +266,53 @@ public void createOAuth2Templates() throws Exception {
...
+        try {
+            log.trace("Executing saveResource [{}]", resource);
+            if (StringUtils.isEmpty(resource.getData())) {
+                throw new DataValidationException("Resource data should be specified!");
+            }
+            toLwm2mResource(resource);
+            resourceService.saveResource(resource);
+        } catch (Exception e) {
+            if (e.getMessage().equals("Resource with such resourceKey already exists!")) {
+                log.trace("File of lwm2m model: [{}]. [{}]", resource.getFileName(), e.getMessage());
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/service/install/InstallScripts.java:311]:
```suggestion
                log.trace("[{}] {}", resource.getFileName(), e.getMessage());
```

File: application/src/main/java/org/thingsboard/server/service/install/InstallScripts.java (hunk 1 above)
Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/service/install/InstallScripts.java:270]: Ok

File: application/src/main/java/org/thingsboard/server/install/ThingsboardInstallService.java (hunk 3 above)
Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/install/ThingsboardInstallService.java:94]: Ok

File: application/src/main/java/org/thingsboard/server/install/ThingsboardInstallService.java (hunk 4 above)
Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/install/ThingsboardInstallService.java:296]: OK

File: application/src/main/java/org/thingsboard/server/service/install/InstallScripts.java (hunk 5 above)
Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/service/install/InstallScripts.java:311]: No

File: application/src/main/java/org/thingsboard/server/service/install/InstallScripts.java [hunk 6]
```diff
@@ -260,4 +266,45 @@ public void createOAuth2Templates() throws Exception {
...
+    private void doSaveLwm2mResource(TbResource resource) throws ThingsboardException {
+        try {
+            log.trace("Executing saveResource [{}]", resource);
+            if (StringUtils.isEmpty(resource.getData())) {
+                throw new DataValidationException("Resource data should be specified!");
+            }
+            toLwm2mResource(resource);
+            resourceService.saveResource(resource);
+        } catch (DataValidationException e) {
+            log.error("[{}] {}", resource.getFileName(), e.getMessage());
```
Comment (ashvayka) [inline application/src/main/java/org/thingsboard/server/service/install/InstallScripts.java:304]: Please use debug here. We don't need these errors in the install log.

Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/service/install/InstallScripts.java]: Ok

**Pull request merged**